nsfc-plugin/src/main/java/org/osc/controller/nsfc/utils/OsCalls.java

```diff
@@ -0,0 +1,230 @@
+
+   public FlowClassifier getFlowClassifier(String flowClassifierId) {
+       return this.osClient.sfc().flowclassifiers().get(flowClassifierId);
+   }
+
+   public PortChain getPortChain(String portChainId) {
+       PortChain portChain = this.osClient.sfc().portchains().get(portChainId);
+       return initializePortChainCollections(portChain);
+   }
+
+   public PortPairGroup getPortPairGroup(String portPairGroupId) {
+       return this.osClient.sfc().portpairgroups().get(portPairGroupId);
+   }
+
+   public PortPair getPortPair(String portPairId) {
+       return this.osClient.sfc().portpairs().get(portPairId);
+   }
+
+   public Port getPort(String portId) {
+       return this.osClient.networking().port().get(portId);
+   }
+
+   public FlowClassifier updateFlowClassifier(String flowClassifierId, FlowClassifier flowClassifier) {
+       checkArgument(flowClassifierId != null, "null passed for %s !", "Flow Classifier Id");
+       checkArgument(flowClassifier != null, "null passed for %s !", "Flow Classifier");
+
+       // OS won't let us modify some attributes. Must be null on update object
+       flowClassifier = flowClassifier.toBuilder().id(null).projectId(null).build();
+
+       return this.osClient.sfc().flowclassifiers().update(flowClassifierId, flowClassifier);
+   }
+
+   public PortChain updatePortChain(String portChainId, PortChain portChain) {
+       checkArgument(portChainId != null, "null passed for %s !", "Port Chain Id");
+       checkArgument(portChain != null, "null passed for %s !", "Port Chain");
+
+       // OS won't let us modify some attributes. Must be null on update object
+       portChain = portChain.toBuilder().id(null).projectId(null).chainParameters(null).chainId(null).build();
+
+       portChain = this.osClient.sfc().portchains().update(portChainId, portChain);
+       return initializePortChainCollections(portChain);
+   }
+
+   public PortPairGroup updatePortPairGroup(String portPairGroupId, PortPairGroup portPairGroup) {
+       checkArgument(portPairGroupId != null, "null passed for %s !", "Port Pair Group Id");
+       checkArgument(portPairGroup != null, "null passed for %s !", "Port Pair Group");
+
+       // OS won't let us modify some attributes. Must be null on update object
+       portPairGroup  = portPairGroup.toBuilder().id(null).projectId(null).portPairGroupParameters(null).build();
+
+       portPairGroup = this.osClient.sfc().portpairgroups().update(portPairGroupId, portPairGroup);
+       return portPairGroup;
+   }
+
+   public PortPair updatePortPair(String portPairId, PortPair portPair) {
+       checkArgument(portPairId != null, "null passed for %s !", "Port Pair Id");
+       checkArgument(portPair != null, "null passed for %s !", "Port Pair");
+
+       // OS won't let us modify some attributes. Must be null on update object
+       portPair = portPair.toBuilder().id(null).projectId(null).build();
+       portPair = this.osClient.sfc().portpairs().update(portPairId, portPair);
```

Comment:
if the port pair is not updated for whatever reason...would it throw a runtime exception? or return null?

If it returns null, we might want to handle and throw exception...same for other places